library/alloc/src/vec/mod.rs

```diff
@@ -2036,6 +2036,64 @@ impl<T, A: Allocator> Vec<T, A> {
         }
     }
 
+    /// Inserts an element at position `index` within the vector, shifting all
+    /// elements after it to the right.
+    ///
+    /// Returns [`None`] if the specified index is out of bounds.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// #![feature(push_mut)]
+    /// let mut vec = vec![1, 3, 5, 9];
+    /// let x = vec.insert_mut(3, 6).unwrap();
+    /// *x += 1;
+    /// assert_eq!(vec, [1, 3, 5, 7, 9]);
+    ///
+    /// let y = vec.insert_mut(7, 6);
+    /// assert!(y.is_none());
+    /// ```
+    ///
+    /// # Time complexity
+    ///
+    /// Takes *O*([`Vec::len`]) time. All items after the insertion index must be
+    /// shifted to the right. In the worst case, all elements are shifted when
+    /// the insertion index is 0.
+    #[cfg(not(no_global_oom_handling))]
+    #[inline]
+    #[unstable(feature = "push_mut", issue = "135974")]
+    #[track_caller]
+    #[must_use = "if you don't need a reference to the value or type-safe bound checking, use Vec::insert instead"]
+    pub fn insert_mut(&mut self, index: usize, element: T) -> Option<&mut T> {
```

Comment:
Was returning an `Option` discussed somewhere? It makes sense to me but I don't see it on the ACPs.

Reply:
It wasn't, it was a pragmatic decision on my part.

Reply:
@rust-lang/libs-api anybody mind giving a +1 or -1 to returning an `Option` rather than panicking?

Reply:
I don't think that @ mention worked.

Reply:
`insert()` also just panics when it's out of bounds, so I don't see a reason for insert_mut to do that differently. I think it should just return `&mut T` rather than `Option<&mut T>`.